### PR TITLE
azurerm_cosmosdb_postgresql_cluster - Mark administrator_login_password as optional

### PR DESCRIPTION
**Pull request by neil-yechenwei**
fixes https://github.com/hashicorp/terraform-provider-azurerm/issues/23734

image
image
image

### CODE REVIEW

**Review by stephybun — CHANGES_REQUESTED** [review]
The password should probably only be optional for read replicas as the issue explains, so could we please add some validation in to make sure it is set if the cluster isn't a replica, i.e. when `source_resource_id` has been omitted.

**Comment by neil-yechenwei** [comment]
@stephybun , thanks for the comment. I updated PR. Please take another look. Thanks.

File: website/docs/r/cosmosdb_postgresql_cluster.html.markdown [hunk 1]
```diff
@@ -49,6 +47,8 @@ The following arguments are supported:
 
 * `node_count` - (Required) The worker node count of the Azure Cosmos DB for PostgreSQL Cluster. Possible value is between `0` and `20` except `1`.
 
+* `administrator_login_password` - (Optional) The password of the administrator login.
```
Comment (stephybun) [inline website/docs/r/cosmosdb_postgresql_cluster.html.markdown:50]: The documentation requires an update too 
```suggestion
* `administrator_login_password` - (Optional) The password of the administrator login. This is required when `source_resource_id` is not set.
```

**Review by stephybun — CHANGES_REQUESTED** [review]

File: website/docs/r/cosmosdb_postgresql_cluster.html.markdown (hunk 1 above)
Comment (neil-yechenwei) [reply website/docs/r/cosmosdb_postgresql_cluster.html.markdown]: Updated

**Comment by neil-yechenwei** [comment]
@stephybun , thanks for the comment. I updated PR. Please take another look. Thanks.

**Review by stephybun — APPROVED** [review]
Thanks @neil-yechenwei LGTM 👍

**Pull request merged**